assets/js/src/segments/dynamic/dynamic_segments_filters/woocommerce.tsx

```diff
@@ -70,21 +85,67 @@ export const WooCommerceFields: React.FunctionComponent<Props> = ({ onChange, it
         </div>
       </div>
     );
-  }
+  } else if (item.action === WooCommerceActionTypes.PURCHASED_CATEGORY) {
+    optionFields = (
+      <div className="mailpoet-form-field">
+        <div className="mailpoet-form-input mailpoet-form-select">
+          <Select
+            placeholder={MailPoet.I18n.t('selectWooPurchasedCategory')}
+            options={categoryOptions}
+            value={find(['value', item.category_id], categoryOptions)}
+            onChange={(option: SelectOption): void => compose([
+              onChange,
+              assign(item),
+            ])({ category_id: option.value })}
+          />
+        </div>
+      </div>
+    );
+  } else if (item.action === WooCommerceActionTypes.NUMBER_OF_ORDERS) {
+    if (!item.number_of_orders_type) {
+      item.number_of_orders_type = '=';
```

Comment:
This assignment produces this error `106:7  error  Assignment to property of function parameter 'item'  no-param-reassign` (https://app.circleci.com/pipelines/github/mailpoet/mailpoet/6280/workflows/f6561af2-88bd-4f95-9026-4259b6e11605/jobs/122376), but I don't know what would be another way to make sure that item.number_of_orders_type is initialized with a default value. I tried setting `defaultValue` inside the Select component and that works for the rendered HTML but it doesn't set the value of item.number_of_orders_type.

Reply:
This doesn't work. You cannot change the `item` properties and expect them to propagate everywhere. The item is defined up in the components tree here: https://github.com/mailpoet/mailpoet/blob/99eff6129ccd96bc11738cb5e6f6b5cf444bd07c/assets/js/src/segments/dynamic/form.tsx#L68

To change the property we need to call the function `setItem` on `DynamicSegmentForm`. That function is passed in this place as `onChange` so you need to call it to update `item`. If you did it in the code here like this: 
`onChange({...})` that would always override the value using the default one. So we need to wrap it in the `useEffect` hook like this:
```
useEffect(() => {
    if (item.number_of_orders_type === undefined) {
      onChange(assign(item, { number_of_orders_type: '=' }));
    }
  }, [onChange, item, item.operator]);
```

Reply:
Thanks for the detailed response. I still need to familiarize myself with all React components. I had also missed where `item` was defined. I pushed a commit implementing your suggestion.